controllers/routemonitor/routemonitor_supplement.go

```diff
@@ -234,11 +234,17 @@ func (r *RouteMonitorReconciler) EnsureRouteURLExists(route routev1.Route, route
 	}
 
 	if currentRouteURL != "" && extractedRouteURL != currentRouteURL {
-		r.Log.V(3).Info("RouteURL mismatch: currentRouteURL and extractedRouteURL are not equal, taking extractedRouteURL as source of truth")
+		err := r.checkRedirect(extractedRouteURL, routeMonitor)
+		if err != nil {
+			r.Log.V(3).Error(err, "RouteURL mismatch: failed to check redirects")
+			return utilreconcile.RequeueReconcileWith(err)
+		} else {
+			r.Log.V(3).Info("RouteURL mismatch: currentRouteURL and extractedRouteURL are not equal, taking extractedRouteURL as source of truth")
```

Comment:
I am sorry, I don't understand your comment. There are two states in which either defaultRouteURL or extractedRouteURL are the source of truth.

Reply:
that's because it doesn't make sense the way I wrote it :D

So when `checkRedirect` doesn't return an error, and finds a redirect, it will log
```
RouteURL mismatch: redirection status code received, taking defaultRouteURL as source of truth
```
in line 278. Since it didn't return an error, this else-branch will be evaluated and log
```
RouteURL mismatch: currentRouteURL and extractedRouteURL are not equal, taking extractedRouteURL as source of truth
```
which seems confusing to me?